jcl/src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -438,18 +438,39 @@ private void readObject(ObjectInputStream s)
 	
 	
 	if (suppressedExceptions != null) {
-		if (suppressedExceptions.size() == 0) {
-			suppressedExceptions = Collections.EMPTY_LIST;
-		} else {
-			for (Throwable t : suppressedExceptions) {
-				if (t == null) {
-					/*[MSG "K0561", "Null entries not permitted in suppressedExceptions serial stream"]*/
-					throw new NullPointerException(com.ibm.oti.util.Msg.getString("K0561")); //$NON-NLS-1$
-				} else if (t == this) {
-					/*[MSG "K0562", "Self-pointers not permitted in suppressedExceptions serial stream"]*/
-					throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0562")); //$NON-NLS-1$
+		List<Throwable> newList = null;
```

Comment:
If a non-null suppressedExceptions cannot be restored, the default value needs to be Collections.EMPTY_LIST

Reply:
fixed